src/zos_files/zowe/zos_files_for_zowe_sdk/datasets.py

```diff
@@ -11,13 +11,215 @@
 """
 
 import os
+from typing import Optional
 
 from zowe.core_for_zowe_sdk import SdkApi
 from zowe.core_for_zowe_sdk.exceptions import FileNotFound
 from zowe.zos_files_for_zowe_sdk.constants import FileType, zos_file_constants
 
 _ZOWE_FILES_DEFAULT_ENCODING = zos_file_constants["ZoweFilesDefaultEncoding"]
 
+
+class DatasetOption:
+
+    def __init__(
+        self,
+        like: Optional[str] = None,
+        volser: Optional[str] = None,
+        unit: Optional[str] = None,
+        dsorg: Optional[str] = None,
+        alcunit: Optional[str] = None,
+        primary: Optional[int] = None,
+        secondary: Optional[int] = None,
+        dirblk: Optional[int] = None,
+        avgblk: Optional[int] = None,
+        recfm: Optional[str] = None,
+        blksize: Optional[int] = None,
+        lrecl: Optional[int] = None,
+        storclass: Optional[str] = None,
+        mgntclass: Optional[str] = None,
+        dataclass: Optional[str] = None,
+        dsntype: Optional[str] = None,
+    ) -> None:
+        self.__like = like
```

Comment:
Should we add a setter for `like` and change this to `self.like = like`?

Reply:
Some of the properties like `alcunit` have different validation depending on whether `like` is set or not.

If we allow the `like` property to be changed, then we need to revalidate these other properties. For now it seemed simpler to only allow setting `like` in the constructor.

Reply:
Ok that makes sense.